shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -244,6 +244,16 @@ - (void)replaceRange:(UITextRange*)range withText:(NSString*)text {
   [self updateEditingState];
 }
 
+- (BOOL)shouldChangeTextInRange:(UITextRange*)range replacementText:(NSString*)text {
+  if (self.returnKeyType == UIReturnKeyDone && [text isEqualToString:@"\n"]) {
```

Comment:
I am not sure how that interacts with the UITextInput protocol. I think this fine though.